File: 392. Is Subsequence/392. Is Subsequence.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isSubsequence = function (s, t) {
  if (s === "") {
    return true;
  }

  const n = s.length;
  const m = t.length;

  if (n > m) {
    return false;
  }

  let i = 0;
  for (let j = 0; j < m; j++) {
    if (s[i] == t[j]) {
      i++;
    }

    if (i == n) {
      return true;
    }
  }

  return false;
};

File: 392. Is Subsequence.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        if s == "":
            return True

        s_len, t_len = len(s), len(t)

        if s_len > t_len:
            return False

        s_pointer = 0

        for t_pointer in range(t_len):
            if s[s_pointer] == t[t_pointer]:
                s_pointer += 1

            if s_pointer == s_len:
                return True

        return False
